src/runtime/test/netprovider.cpp

```diff
@@ -272,9 +272,12 @@ TEST(tools_common, asio_network_provider_connection_threshold)
     error_code start_result;
     start_result = asio_network->start(RPC_CHANNEL_TCP, TEST_PORT, false);
     ASSERT_TRUE(start_result == ERR_OK);
-    asio_network->change_test_cfg_conn_threshold_per_ip(10);
 
-    for (int count = 0; count < 20; count++) {
+    auto CONN_THRESHOLD = 3;
+    asio_network->change_test_cfg_conn_threshold_per_ip(CONN_THRESHOLD);
+
+    // not exceed threshold
+    for (int count = 0; count < CONN_THRESHOLD + 2; count++) {
```

Comment:
Why doesn't it exceed threshold when sessions count == CONN_THRESHOLD +2?

Reply:
Acutually, the count doesn't matter in not-exceed case, it will first create a client to create a new session, then close the session, so it will not exceed connection threshold, the count here only means the loop count.